eclair-core/src/test/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisherSpec.scala

```diff
@@ -1220,12 +1308,12 @@ class ReplaceableTxPublisherSpec extends TestKitBaseClass with AnyFunSuiteLike w
     generateBlocks(1)
 
     alice2blockchain.expectMsgType[PublishFinalTx] // claim main output
-    val claimHtlcTimeout = alice2blockchain.expectMsgType[PublishReplaceableTx]
-    assert(claimHtlcTimeout.txInfo.isInstanceOf[ClaimHtlcTimeoutTx])
-    val claimHtlcTimeoutTx = claimHtlcTimeout.txInfo.asInstanceOf[ClaimHtlcTimeoutTx].copy(confirmBefore = overrideHtlcTarget)
     val claimHtlcSuccess = alice2blockchain.expectMsgType[PublishReplaceableTx]
     assert(claimHtlcSuccess.txInfo.isInstanceOf[ClaimHtlcSuccessTx])
     val claimHtlcSuccessTx = claimHtlcSuccess.txInfo.asInstanceOf[ClaimHtlcSuccessTx].copy(confirmBefore = overrideHtlcTarget)
+    val claimHtlcTimeout = alice2blockchain.expectMsgType[PublishReplaceableTx]
```

Comment:
I tried to understand why the order in which we check for `ClaimHtlcSuccessTx` and `ClaimHtlcTimeoutTx` and I'm not sure there is a deterministic order we can rely upon since we use unordered collections (sets, maps,...) when we generate and publish these transactions ?

Reply:
I checked, and in general there is no order guarantee for Sets. But for small sets (with less than 4 elements) Scala uses ad-hoc implementations where indeed items are kept in the order they were added (see https://github.com/scala/scala/blob/e41a0cc980012d20874d2191b5aef6fb78a7cf27/src/library/scala/collection/immutable/Set.scala#L183). 
We should add a comment in the tests next time they're modified.